src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -318,28 +330,43 @@ protected function instantiateObject(array &$data, $class, array &$context, \Ref
 
                         $params = array_merge($params, $data[$paramName]);
                     }
-                } elseif ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
-                    $params[] = $data[$key];
-                    // don't run set for a parameter passed to the constructor
+
+                    continue;
+                }
+
+                if ($allowed && !$ignored && (isset($data[$key]) || array_key_exists($key, $data))) {
+                    $parameterData = $data[$key];
+                    if (null !== $constructorParameter->getClass()) {
```

Comment:
In case you use a type hint with a class from a third party library which is not loaded because you don't need it for example.
And the advantage of getType() is that it doesn't need to load the class, so would work anyway.